webapp/channels/src/components/plugin_link_tooltip/index.tsx

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {
+    useFloating,
+    autoUpdate,
+    safePolygon,
+    useHover,
+    useDismiss,
+    useInteractions,
+    FloatingPortal,
+    autoPlacement,
+    inline,
+    useTransitionStyles,
+    FloatingOverlay,
+    FloatingFocusManager,
+    arrow,
+    offset,
+    FloatingArrow,
+    useFocus,
+    useRole,
+} from '@floating-ui/react';
+import classNames from 'classnames';
+import React, {useRef, useState} from 'react';
+import type {AnchorHTMLAttributes, ReactElement} from 'react';
+
+import Pluggable from 'plugins/pluggable';
+import {RootHtmlPortalId, OverlaysTimings, OverlayArrow, A11yClassNames} from 'utils/constants';
+
+import './plugin_link_tooltip.scss';
+
+interface Props {
+    nodeAttributes: AnchorHTMLAttributes<HTMLAnchorElement>;
+    children: ReactElement;
+}
+
+function PluginLinkTooltip(props: Props) {
+    const [isOpen, setOpen] = useState(false);
+
+    const arrowRef = useRef(null);
+
+    const {refs: {setReference, setFloating}, floatingStyles, context: floatingContext} = useFloating({
+        open: isOpen,
+        onOpenChange: setOpen,
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offset(OverlayArrow.OFFSET),
+            inline(),
+            autoPlacement({
+                allowedPlacements: ['top', 'bottom'],
+            }),
+            arrow({
+                element: arrowRef,
+            }),
+        ],
+    });
+
+    const {isMounted, styles: transitionStyles} = useTransitionStyles(floatingContext, {
+        duration: {
+            open: OverlaysTimings.FADE_IN_DURATION,
+            close: OverlaysTimings.FADE_OUT_DURATION,
+        },
+        initial: {
+            opacity: 0,
+        },
+        common: {
+            opacity: 1,
+        },
+    });
+
+    const combinedFloatingStyles = Object.assign({}, floatingStyles, transitionStyles);
+
+    const hoverInteractions = useHover(floatingContext, {
+        restMs: OverlaysTimings.CURSOR_REST_TIME_BEFORE_OPEN,
+        move: false,
+        handleClose: safePolygon({
+            requireIntent: false,
+            blockPointerEvents: true,
+        }),
+    });
+    const focusInteractions = useFocus(floatingContext);
+    const dismissInteraction = useDismiss(floatingContext);
+    const roleProps = useRole(floatingContext, {role: 'tooltip'});
+
+    const {getReferenceProps, getFloatingProps} = useInteractions([
+        hoverInteractions,
+        focusInteractions,
+        dismissInteraction,
+        roleProps,
+    ]);
+
+    return (
+        <>
+            <a
+                ref={setReference}
+                {...props.nodeAttributes}
+                {...getReferenceProps()}
```

Comment:
Since we're focused on the memory issues with this change, this may be out of scope, but will this cause all links in posts to be given ARIA attributes like `aria-haspopup` when the Jira or GH plugins are enabled, even if the link doesn't have a tooltip?

If so, I think we'll want to manually remove those properties until we can change the plugin API for this. It'd be better to keep the current behaviour than to add misleading ARIA attributes here

Reply:
agreed, will remove

Reply:
verified and removed the hook which was adding the aria-* attrs to this without manual removing from here